src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs

```diff
@@ -16,22 +14,9 @@ internal partial class InteractiveWorkspace : Workspace
     private SourceTextContainer? _openTextContainer;
     private DocumentId? _openDocumentId;
 
-    internal InteractiveWorkspace(HostServices hostServices, IGlobalOptionService globalOptions)
+    internal InteractiveWorkspace(HostServices hostServices)
         : base(hostServices, WorkspaceKind.Interactive)
     {
-        // register work coordinator for this workspace
-        if (globalOptions.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
-        {
-            Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Register(this);
-        }
-    }
-
-    protected override void Dispose(bool finalize)
-    {
-        // workspace is going away. unregister this workspace from work coordinator
-        Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Unregister(this, blockingShutdown: true);
-
-        base.Dispose(finalize);
```

Comment:
no solution crawler, so workspaces are now decoupled from the deeply embedded concept.